pkg/core/blockchain.go

```diff
@@ -3130,7 +3158,22 @@ func (bc *Blockchain) newInteropContext(trigger trigger.Type, d *dao.Simple, blo
 		// changes that were not yet persisted to Blockchain's dao.
 		baseStorageFee = bc.contracts.Policy.GetStoragePriceInternal(d)
 	}
-	ic := interop.NewContext(trigger, bc, d, baseExecFee, baseStorageFee, native.GetContract, bc.contracts.Contracts, contract.LoadToken, block, tx, bc.log)
+	var (
+		maxVUBIncrement = bc.config.Genesis.MaxValidUntilBlockIncrement
+		hf              = config.HFEchidna
+		index           uint32
+	)
+	if block != nil {
+		index = block.Index
+	} else {
+		index = bc.BlockHeight() + 1 // mock persisting block index.
+	}
+	if bc.isHardforkEnabled(&hf, index) && !bc.isHardforkActivationHeight(&hf, index) { // Policy setting is accepted starting from Echidna+1 height.
+		// Use provided dao instead of Blockchain's one to fetch possible MaxVUBIncrement
+		// changes that were not yet persisted to Blockchain's dao.
+		maxVUBIncrement = bc.contracts.Policy.GetMaxValidUntilBlockIncrementInternal(d)
+	}
+	ic := interop.NewContext(trigger, bc, d, baseExecFee, baseStorageFee, maxVUBIncrement, native.GetContract, bc.contracts.Contracts, contract.LoadToken, block, tx, bc.log)
```

Comment:
It's not the most elegant solution, may be it's better this HF-dependent switch directly to interop context, I'll check it.

Reply:
Fixed, I decided to bind it directly to Policy since it's the Policy contract who is responsible for the method enabling logic.